internal/js_parser/js_parser_test.go

```diff
@@ -5285,3 +5285,23 @@ func TestAutoPureForWeakMap(t *testing.T) {
 	expectPrinted(t, "new WeakMap([x, []])", "new WeakMap([x, []]);\n")
 	expectPrinted(t, "new WeakMap([[], x])", "new WeakMap([[], x]);\n")
 }
+
+func TestDropLocalUnusedVariable(t *testing.T) {
+	expectPrintedMangle(t, "var v = 0", "var v = 0;\n")
+	expectPrintedMangle(t, "let v = 0", "let v = 0;\n")
+	expectPrintedMangle(t, "const v = 0", "const v = 0;\n")
+	expectPrintedMangle(t, "{ var v = 0 }", "var v = 0;\n")
+	expectPrintedMangle(t, "{ let v = 0 }", "")
+	expectPrintedMangle(t, "{ let v = 3; let v2 = v + 5 }", "{\n  let v2 = 3 + 5;\n}\n") // does not support multi-level
+	expectPrintedMangle(t, "{ const v = 3; const v2 = v + 5 }", "")
+	expectPrintedMangle(t, "{ const v = 0 }", "")
+	expectPrintedMangle(t, "{ const v = (() => 'foo')() }", "{\n  const v = (() => \"foo\")();\n}\n")
+	expectPrintedMangle(t, "{ const v = /* #__PURE__ */ (() => 'foo')() }", "")
+	expectPrintedMangle(t, "{ const v = 0; console.log(v) }", "console.log(0);\n")
+	// expectPrintedMangle(t, "function f() { const v = 0; eval('v') }", "function f() {\n  const v = 0;\n  eval(\"v\");\n}\n")
```

Comment:
This test fails without this PR.
`const v = 0;` is dropped at here.
https://github.com/evanw/esbuild/blob/698898f86615a69f52d2e00681a13fe2b8635e88/internal/js_parser/js_parser.go#L9312-L9317